cpacs2to3/material.py

```diff
@@ -0,0 +1,437 @@
+                "Maybe you intended to create material properties for UD-CFK. "
+                "Then use transversal isotrop material instead!"
+            )
+            log.warning(warnMsg)
+
+        else:
+            if tixi.checkElement(xPath + "/k23"):  # required element for transversal isotrop material
+                # transversal isotrop material
+                for stiffEntry in ["k11", "k12", "k22", "k23", "k66"]:
+                    stiffnessDict[stiffEntry] = tixi.getDoubleElement(xPath + "/" + stiffEntry)
+
+                sig11t = tixi.getDoubleElement(xPath + "/sig11t")
+                sig11c = tixi.getDoubleElement(xPath + "/sig11c")
+                sig22t = tixi.getDoubleElement(xPath + "/sig22t")
+                sig22c = tixi.getDoubleElement(xPath + "/sig22c")
+                tau12 = tixi.getDoubleElement(xPath + "/tau12")
+                tau23 = tixi.getDoubleElement(xPath + '/tau23')
+
+            elif tixi.checkElement(xPath + "/k11"):  # required element for isotrop material
+                # isotrop material
+                for stiffEntry in ["k11", "k12"]:
+                    stiffnessDict[stiffEntry] = tixi.getDoubleElement(xPath + "/" + stiffEntry)
+
+                sig11t = tixi.getDoubleElement(xPath + "/sig11")
+                tau12 = tixi.getDoubleElement(xPath + "/tau12")
+
+                # creating stiffness params for transversal isotrop material
+                stiffnessDict["k22"] = stiffnessDict["k11"]
+                stiffnessDict["k23"] = stiffnessDict["k12"]
+                stiffnessDict["k66"] = 0.5 * (stiffnessDict["k11"] - stiffnessDict["k12"])
+
+                sig11c, sig22t, sig22c = [sig11t] * 3
+                tau23 = tau12
+            else:
+                raise ValueError("wrong material definition at xPath " + xPath)
+
+            # creating stiffness params for orthotrop material
+            stiffnessDict["k13"] = stiffnessDict["k12"]
+            stiffnessDict["k33"] = stiffnessDict["k22"]
+            stiffnessDict["k44"] = 0.5 * (stiffnessDict["k22"] - stiffnessDict["k23"])
+            stiffnessDict["k55"] = stiffnessDict["k66"]
+
+        # now each stiffness needed for an orthotrop definition is known
+        for i in range(6):
+            self.stiffnessMatrix[i, i] = stiffnessDict["k%s%s" % (i + 1, i + 1)]
+
+        self.stiffnessMatrix[0, 2] = stiffnessDict["k13"]
+        self.stiffnessMatrix[2, 0] = stiffnessDict["k13"]
+
+        self.stiffnessMatrix[0, 1] = stiffnessDict["k12"]
+        self.stiffnessMatrix[1, 0] = stiffnessDict["k12"]
+
+        self.stiffnessMatrix[2, 1] = stiffnessDict["k23"]
+        self.stiffnessMatrix[1, 2] = stiffnessDict["k23"]
+
+        self.strength["sigma11t"] = abs(sig11t)
+        self.strength["sigma22t"] = abs(sig22t)
+        self.strength["sigma11c"] = abs(sig11c)
+        self.strength["sigma22c"] = abs(sig22c)
+        self.strength["tau12"] = tau12
+        self.strength["tau23"] = tau23
```

Comment:
tau23 might be invalid / uninitialzed, if the first if clause is true.

Reply:
adjusted